packages/back-end/src/jobs/updateExposureQueriesDimensionSlices.ts

```diff
@@ -0,0 +1,266 @@
+import {
+  _dangerousGetAllDatasourcesWithExposureQueriesWithAutomaticDimensionSlices,
+  getDataSourceById,
+  updateDataSource,
+} from "back-end/src/models/DataSourceModel";
+import { usingFileConfig } from "back-end/src/init/config";
+import { getIntegrationFromDatasourceId } from "back-end/src/services/datasource";
+import {
+  createDimensionSlices,
+  _dangerousGetAllDimensionSlicesByIds,
+} from "back-end/src/models/DimensionSlicesModel";
+import { DimensionSlicesQueryRunner } from "back-end/src/queryRunners/DimensionSlicesQueryRunner";
+
+const QUEUE_EXPOSURE_QUERIES_DIMENSION_SLICES_UPDATES =
+  "queueExposureQueriesDimensionSlicesUpdates";
+
+const UPDATE_SINGLE_DATASOURCE_EXPOSURE_QUERIES_DIMENSION_SLICES =
+  "updateSingleExposureQueryDimensionSlices";
+
+const DEFAULT_LOOKBACK_DAYS = 30;
+
+type UpdateSingleJobParams = {
+  organizationId: string;
+  dataSourceId: string;
+  exposureQueryIds: string[];
+};
+
+export default async function (agenda: Agenda) {
+  agenda.define(QUEUE_EXPOSURE_QUERIES_DIMENSION_SLICES_UPDATES, async () => {
+    const exposureQueriesToUpdate =
+      await getDataSourcesWithExposureQueriesToUpdate();
+
+    for (const jobParams of exposureQueriesToUpdate) {
+      await queueExposureQueriesDimensionSlicesUpdate(jobParams);
+    }
+  });
+
+  agenda.define(
+    UPDATE_SINGLE_DATASOURCE_EXPOSURE_QUERIES_DIMENSION_SLICES,
+    updateSingleDatasourceExposureQueriesDimensionSlices,
+  );
+
+  // Don't schedule automatic updates if using file config
+  if (usingFileConfig()) return;
+
+  await startUpdateJob();
+
+  async function startUpdateJob() {
+    const job = agenda.create(
+      QUEUE_EXPOSURE_QUERIES_DIMENSION_SLICES_UPDATES,
+      {},
+    );
+    job.unique({});
+    job.repeatEvery("1 day");
+    await job.save();
+  }
+
+  async function queueExposureQueriesDimensionSlicesUpdate(
+    jobParams: UpdateSingleJobParams,
+  ) {
+    logger.info(
+      {
+        dataSourceId: jobParams.dataSourceId,
+        exposureQueryIds: jobParams.exposureQueryIds,
+      },
+      "Queuing dimension slices update",
+    );
+
+    const job = agenda.create(
+      UPDATE_SINGLE_DATASOURCE_EXPOSURE_QUERIES_DIMENSION_SLICES,
+      jobParams,
+    );
+    job.unique({
+      organizationId: jobParams.organizationId,
+      dataSourceId: jobParams.dataSourceId,
+    });
+    job.schedule(new Date());
+    await job.save();
+  }
+}
+
+export async function getDataSourcesWithExposureQueriesToUpdate(): Promise<
+  UpdateSingleJobParams[]
+> {
+  try {
+    const potentialDataSourcesToUpdate =
+      await _dangerousGetAllDatasourcesWithExposureQueriesWithAutomaticDimensionSlices();
+
+    // Include only exposure queries that have at least one non-custom slice
+    const dataSourcesAndExposureQueries = potentialDataSourcesToUpdate
+      .map((ds) => {
+        const exposureQueries = ds.settings?.queries?.exposure || [];
+        const queriesWithAutomaticDimensionSlices = exposureQueries.filter(
```

Comment:
If an exposure query has no `dimensionMetadata` but they do not have `disablePrecomputedDimensions` we should consider updating for a kind of "auto-onboarding" to pre-computed dimensions.

Maybe, however, there should be a separate onboarding flow that signals they want this automatic job to run?